Gallery edit page

Jessonsotoventura opened the pull request:
Gallery Edit page first draft by @agarwali

agarwali committed:
added views for gallery edit page

agarwali committed:
added views for gallery edit page

agarwali committed:
Merge remote-tracking branch 'origin/master' into galleryEditPage

agarwali committed:
added wysiwig layout and dropzone

agarwali committed:
added wysiwig layout and dropzone

memo330179 commented on app/config/secret.yaml:
```diff
@@ -2,6 +2,6 @@ db:
-  username: 
-secret_key: secret_key
+  password: ''
+  username: agarwali
```
We should not be committing this file. We will add it to the `.gitignore`

memo330179 commented on app/templates/snips/base.html:
```diff
@@ -41,6 +41,9 @@
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datetimepicker/4.17.37/css/bootstrap-datetimepicker.min.css"/>
     
+    <!--Ckeditor-->
+    <script src="/static/js/ext/ckeditor/ckeditor.js"></script>
```
This should be moved to specific template to avoid always loading

memo330179 commented on app/templates/views/gallery_edit.html:
```diff
@@ -1,4 +1,64 @@
+  </div>
+  <script type="text/javascript">
+            $(function () {
+                $('#startDate').datetimepicker();
```
Javascript should be moved to it's own file `app/static/js/local/filename`.

memo330179 commented on app/templates/views/gallery_edit.html:
```diff
@@ -1,4 +1,64 @@
+  </div>
+  <script type="text/javascript">
+            $(function () {
+                $('#endDate').datetimepicker();
```
Javascript should be moved to its own file app/static/js/local/filename.

memo330179 commented on app/templates/views/gallery_edit.html:
```diff
@@ -1,4 +1,64 @@
+    <script>
+        // Replace the <textarea id="editor1"> with a CKEditor
+        // instance, using default configuration.
+        CKEDITOR.replace( 'editor1' );
```
Javascript should be moved to its own file app/static/js/local/filename.

memo330179 reviewed (changes requested):
This looks good. Just noted some changes.

agarwali committed:
Merge remote-tracking branch 'origin/master' into galleryEditPage

memo330179 commented:
@agarwali I just realized you probably don't get notifications to this PR.

agarwali committed:
merged development

Commit:
Simplified running locally

- Added an option to add a custom path into the program
- Added an option to add the current path into the program

agarwali committed:
Edit backend fully working except upload

Added the backend of gallery edit page and created a new controller for add gallery. The file upload is breaking so I have temporarily removed it.

agarwali committed:
Edit backend fully working except upload

Added the backend of gallery edit page and created a new controller for add gallery. The file upload is breaking so I have temporarily removed it.

agarwali committed:
Merge remote-tracking branch 'origin/development' into galleryEditPage

memo330179 reviewed (dismissed):
We should not bloat the repo by including dependencies. We should either use NPM, wget, or some other method of installation during the application setup.

memo330179 committed:
Merge pull request #25 from BCStudentSoftwareDevTeam/local_app_run_improvement

Simplify Running the app locally

Jessonsotoventura commented:
I think @agarwali has a good reason for adding in the dependency. I believe that if we stray from the default ckeditor build it makes it hard to include using NPM or wget, since the config files change. In this case I think we may end up changing the standard button config, but we will certainly change the theme to bootstrap. @memo330179 at least using the a custom theme requires a custom build

Jessonsotoventura commented on app/logic/validation.py:
```diff
@@ -35,6 +35,7 @@ def doesUserHaveRole(role):
   username = getUsernameFromEnv()
   try:
     user = UserQueries.select_single(username)
+    print ("Query working", user)
```
Can we remove this print if the function works? 
I think it will get bothersome to have so many prints in the console.

Jessonsotoventura commented on app/models/queries/GalleryQueries.py:
```diff
@@ -14,3 +14,56 @@ def get(gid):
         if Galleries.select().where(Galleries.gid == gid).exists():
             return Galleries.get(Galleries.gid == gid)
     return None
+
```
Thanks for creating and commenting on these, they look good 👍

Jessonsotoventura commented on app/static/js/local/gallery_edit.js:
```diff
@@ -0,0 +1,26 @@
+CKEDITOR.replace( 'description', {
+    customConfig: '/static/js/ckeditor/custom_config.js'
```
Going back to @memo330179's comment could we use a base install say from a CDN and use this to load the custom parts  like the bootstrap theme?

Commit:
Merge branch 'galleryEditPage' of github.com:BCStudentSoftwareDevTeam/Project_ART into galleryEditPage

Commit:
Slight changes to the formatting in the html

Looks good, nice job Ishwar
